### PR TITLE
Run 3 Analysis Setup

### PR DESCRIPTION
**Pull request by haddadanas**
This PR adds the configuration files for run 3 and adjusts some functions of the analysis for it :partying_face: 

New:
- New Correction files
- Separate tau trigger from tau correction factors
- Adjusted the input parameters of the `correctionlib`
-  Added new 2022 Triggers and adjusted the analysis for the new cross `tau-tau-jet` trigger
- Added new Triggers for 2023 (however not tested yet)
- Refactor all working points into the config files

### CODE REVIEW

File: hbt/calibration/default.py [hunk 1]
```diff
@@ -42,7 +45,10 @@ def default(self: Calibrator, events: ak.Array, **kwargs) -> ak.Array:
         events = self[jec_full](events, **kwargs)
         events = self[jer](events, **kwargs)
 
-    events = self[met_phi](events, **kwargs)
+    if self.config_inst.has_tag("run2"):
+        events = self[met_phi](events, **kwargs)
+    else:
+        print("met_phi_setup will be skipped for now, since correction files for run 3 are not yet available!")
```
Comment (riga) [inline hbt/calibration/default.py:51]: We can drop this case as there will likely be no met phi correction.

File: hbt/config/configs_run2ul.py [hunk 2]
```diff
@@ -48,7 +48,7 @@ def add_config(
 
     # create a config by passing the campaign, so id and name will be identical
     cfg = analysis.add_config(campaign, name=config_name, id=config_id)
-
+    cfg.add_tag("run2")
```
Comment (riga) [inline hbt/config/configs_run2ul.py:51]: There is an aux field "run" now directly on the campaign that we should use.

File: hbt/production/tau.py [hunk 3]
```diff
@@ -98,30 +98,42 @@ def tau_weights(self: Producer, events: ak.Array, **kwargs) -> ak.Array:
...
+
+    mu_args = lambda mask, wp, syst: (abseta[mask], match[mask], wp, syst)
 
     # genuine taus
     tau_mask = flat_np_view(dm_mask & (events.Tau.genPartFlav == 5), axis=1)
     sf_nom[tau_mask] = self.id_vs_jet_corrector(*tau_args(tau_mask, "nom"))
 
     # electrons faking taus
     e_mask = ((events.Tau.genPartFlav == 1) | (events.Tau.genPartFlav == 3))
+    if self.config_inst.has_tag("run3"):
```
Comment (riga) [inline hbt/production/tau.py:124]: Should use the "run" aux entry of the campaign.

**Review by riga — CHANGES_REQUESTED** [review]
Awesome work!
I just have a handful of inline comments that should be easy to adapt, but we can also discuss them in person.
🎉

File: hbt/config/configs_run3.py [hunk 4]
```diff
@@ -0,0 +1,803 @@
...
+    if campaign.x.year == 2024:
+        raise NotImplementedError("It a bit too early for 2024 analysis :)")
+
+    # gather campaign data
+    year = campaign.x.year
+    year2 = year % 100
+
+    # postfix for 2022 campaigns after ECAL Endcap water leak
+    year_postfix = "EE" if campaign.x.postfix == "post" else ""
+    postfixEE = f"{campaign.x.postfix}EE"
```
Comment (riga) [inline hbt/config/configs_run3.py:47]: `postfix_ee`

File: hbt/config/configs_run3.py [hunk 5]
```diff
@@ -0,0 +1,803 @@
...
+
+    # get all root processes
+    procs = get_root_processes_from_campaign(campaign)
+
+    # create a config by passing the campaign, so id and name will be identical
+    cfg = analysis.add_config(campaign, name=config_name, id=config_id)
+    cfg.add_tag("run3")
```
Comment (riga) [inline hbt/config/configs_run3.py:54]: We can skip this when we use the campaigns' "run" aux field.

File: hbt/config/configs_run3.py [hunk 6]
```diff
@@ -0,0 +1,803 @@
...
+        cfg.add_process(procs.get(process_name))
+
+        # # add the process (and set xsec to 0.1 if not available)
+        # process_inst = procs.get(process_name)
+
+        # cfg.add_process(process_inst)
+        # for proc_inst in cfg.get_process(process_inst).get_leaf_processes():
+        #     if campaign.ecm not in proc_inst.xsecs.keys():
+        #         cfg.get_process(proc_inst.name).set_xsec(campaign.ecm, Number(0.1))
+
```
Comment (riga) [inline hbt/config/configs_run3.py:92]: We can remove these lines.

File: hbt/config/configs_run3.py [hunk 7]
```diff
@@ -0,0 +1,803 @@
...
+        "TimePtEta",
+    ]
+
+    # name of the btag_sf correction set and jec uncertainties to propagate through
+    cfg.x.btag_sf = ("particleNet_shape", cfg.x.btag_sf_jec_sources, "btagPNetB")
+
+    # name of the deep tau tagger
+    # (used in the tec calibrator)
+    # https://twiki.cern.ch/twiki/bin/view/CMS/TauIDRecommendationForRun3
+    cfg.x.tau_tagger = "DeepTau2018v2p5"  # "DeepTauv2p5"
```
Comment (riga) [inline hbt/config/configs_run3.py:404]: We can remove the comment :)

File: hbt/config/configs_run3.py [hunk 8]
```diff
@@ -0,0 +1,803 @@
...
+
+    # name of the MET phi correction set
+    # (used in the met_phi calibrator)
+    # cfg.x.met_phi_correction_set = "{variable}_metphicorr_pfmet_{data_source}"
```
Comment (riga) [inline hbt/config/configs_run3.py:408]: Can also be removed.

File: hbt/config/configs_run3.py [hunk 9]
```diff
@@ -0,0 +1,803 @@
...
+        "murmuf",
+        {
+            "murmuf_weight": "murmuf_weight_{direction}",
+            "normalized_murmuf_weight": "normalized_murmuf_weight_{direction}",
+        },
+    )
+
+    # external files
+    json_mirror = "/afs/cern.ch/user/a/anhaddad/public/jsonpog-integration"
+    json_mirror_alt = "/afs/cern.ch/user/a/anhaddad/public/jsonpog_alt"
```
Comment (riga) [inline hbt/config/configs_run3.py:589]: `json_mirror = "/afs/cern.ch/user/m/mrieger/public/mirrors/jsonpog-integration-7d9636bf"`
:)

Then we can also change `json_mirror_alt` to `json_mirror` below.

File: hbt/config/configs_run3.py [hunk 10]
```diff
@@ -0,0 +1,803 @@
...
+                # Problem No file for 2022 --> using 2023 no matching shapes with root shape
+                "mc_profile": ("https://raw.githubusercontent.com/cms-sw/cmssw/203834e3ae301f2564423dd1cc84bebf660519b9/SimGeneral/MixingModule/python/Run3_2022_LHC_Simulation_10h_2h_cfi.py", "v1"),  # noqa
+                "data_profile": {
+                    "nominal": (f"/afs/cern.ch/user/a/anhaddad/public/Collisions22/pileupHistogram-Cert_Collisions2022_355100_362760_GoldenJson-13p6TeV-69200ub-100bins.root", "v1"),  # noqa
+                    "minbias_xs_up": (f"/afs/cern.ch/user/a/anhaddad/public/Collisions22/pileupHistogram-Cert_Collisions2022_355100_362760_GoldenJson-13p6TeV-72400ub-100bins.root", "v1"),  # noqa
+                    "minbias_xs_down": (f"/afs/cern.ch/user/a/anhaddad/public/Collisions22/pileupHistogram-Cert_Collisions2022_355100_362760_GoldenJson-13p6TeV-66000ub-100bins.root", "v1"),  # noqa
+                },
+            },
+        }))
+    else:  # year 2023
```
Comment (riga) [inline hbt/config/configs_run3.py:654]: Explicit `elif year == 2023` would be better here.

**Review by riga — CHANGES_REQUESTED** [review]
Forgot to review the config itself ... super impressive.

**Review by pkausw — CHANGES_REQUESTED** [review]
Thanks for all the work @haddadanas ! Just some minor comments from my side

File: hbt/config/configs_run3.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (haddadanas) [reply hbt/config/configs_run3.py:589]: sure, we can do it tomorrow

**Comment by haddadanas** [comment]
Thank you @riga and @pkausw for the feedback! 
I corrected most of it, but there are still some unresolved conversations, which I'd like to discuss with you in person :)

**Comment by haddadanas** [comment]
Done :heavy_check_mark: 
Well almost. Only thing missing is to replace the link of `json_mirror_alt`  to a link on Marcel's afs (I left the comment unresolved)

File: hbt/config/configs_run3.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (riga) [reply hbt/config/configs_run3.py]: You can change this to `/afs/cern.ch/work/m/mrieger/public/mirrors/jsonpog-integration-taupog`, I copied your directory there.

**Review by riga — APPROVED** [review]

**Pull request merged**